trpcultivate_germcollection/src/Plugin/TripalImporter/GermplasmRelationshipImporter.php

```diff
@@ -0,0 +1,1193 @@
+    ];
+
+    $this->importPopulation($population);
+  }
+
+  /**
+   * Function callback, create population.
+   *
+   * @param array $population
+   *   Array, with the following keys:
+   *   entry: Form field value for Primary Germplasm field.
+   *   verb : Form field value for Relationship Verb field.
+   *   position: Form field value for Stock Position field.
+   *   individuals: Form file field value for Related Germplasm Field.
+   */
+  public function importPopulation($population) {
+    $file = $this->service_entityTypeManager->getStorage('file')->load($population['individuals']);
+    $this->setTotalItems($file->filesize);
+    $this->setItemsHandled(0);
+    // Get the mime type which is used to validate the file and split the rows.
+    $file_mime_type = $file->getMimeType();
+
+    $temp_lines = [];
+    $temp_uname = [];
+    $duplicate  = [];
+
+    if ($file) {
+      $file_uri = $file->getFileUri();
+      $handle = fopen($file_uri, 'r');
+      if ($handle) {
+        $i = 0;
+
+        while ($cur_line = fgets($handle)) {
+          // Add all individuals in file into stock table
+          // and simultaneously creating the Relationship verb.
+          if ($i == 0) {
+            // This is the header row.
+            $i++;
+            continue;
+          }
+
+          // Data rows.
+          if ($cur_line) {
+            $data_row = ImportValidationHelper::splitRowIntoColumns($cur_line, $file_mime_type);
+            $val_name = $data_row[0];
+            $val_type = $data_row[1];
+            $val_sciname = $data_row[2];
+            $val_uniqname = $data_row[3] ?? NULL;
+
+            if (!$population['relationship_only']) {
+              if (!$val_uniqname) {
+                throw new \Exception('Uniquename is required in line #' . ($i + 1) . ' when Related germplasm must already exist option is selected.');
+              }
+              $uniquename = $val_uniqname;
+            }
+            else {
+              $uniquename = $val_uniqname;
+            }
+
+            // Organism:
+            // If Scientific Name is present, lookup the organism ID.
+            if ($val_sciname) {
+              $organism_id = $this->getOrganismIds($val_sciname);
+            }
+            else {
+              $organism_id = NULL;
+            }
+            // Throw exception if organism is not valid.
+            if ($organism_id == NULL) {
+              throw new \Exception('Scientific Name: ' . $val_sciname . ' is not valid. Please provide a valid Scientific Name.');
+            }
+
+            // Type:
+            // Use the database name and cvterm encoded in Type to determine
+            // the stock type.
+            if ($val_type) {
+              $type_id = ChadoCVTermAutocompleteController::getCVtermId($val_type);
+            }
+            else {
+              $type_id = NULL;
+            }
+
+            // Throw exception if type is not valid.
+            if ($type_id == NULL) {
```

Comment:
This should actually check the type_id is an integer and > 0. Currently it will catch the case where the type is not in the file but it won't catch the case where `getCVtermId()` is not able to find a type (i.e. returns `0`).